### ISSUE TITLE
Weird optimization behaviour using the TimeDistributed layer

### ISSUE THREAD

**Issue by borismul** [keras-team/keras#11862]
Please make sure that the boxes below are checked before you submit your issue.
If your issue is an **implementation question**, please ask your question on [StackOverflow](http://stackoverflow.com/questions/tagged/keras) or [on the Keras Slack channel](https://keras-slack-autojoin.herokuapp.com/) instead of opening a GitHub issue.

Thank you!

- [x] Check that you are up-to-date with the master branch of Keras. You can update with:
`pip install git+git://github.com/keras-team/keras.git --upgrade --no-deps`

- [x] Check that your version of TensorFlow is up-to-date. The installation instructions can be found [here](https://www.tensorflow.org/get_started/os_setup).

- [x] Provide a link to a GitHub Gist of a Python script that can reproduce your issue (or just copy the script here if it is short).

Hi,

I was using a TimeDistributed layer for an autoencoder for another project of mine. I got some weird results so I checked with the basic example provided in the attachement. I have provided 2 cases which, in my opinion, should be equivalent. However, the loss optimzation is completely different and much more unstable for the TimeDistributed version of the autoencoder. Am I doing something wrong or is this a problem under the hood of Keras?

P.S. 
I'm using tensorflow-gpu as my backend.

[MNIST_TestCase.zip](https://github.com/keras-team/keras/files/2676293/MNIST_TestCase.zip)
